site/src/modules/notifications/NotificationsInbox/NotificationsInbox.stories.tsx

```diff
@@ -0,0 +1,178 @@
+		),
+	},
+};
+
+export const Failure: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: fn(() =>
+			Promise.reject(
+				mockApiError({
+					message: "Failed to load notifications",
+				}),
+			),
+		),
+	},
+};
+
+export const FailAndRetry: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: (() => {
+			let count = 0;
+
+			return fn(() => {
+				count += 1;
+
+				// Fail on the first 3 attempts
+				// 3 is the maximum number of retries from react-query
+				if (count < 3) {
+					return Promise.reject(
+						mockApiError({
+							message: "Failed to load notifications",
+						}),
+					);
+				}
+
+				return Promise.resolve({
+					notifications: MockNotifications,
+					unread_count: 2,
+				});
+			});
+		})(),
+	},
+	play: async ({ canvasElement }) => {
+		const body = within(canvasElement.ownerDocument.body);
+		await expect(
+			body.getByText("Error loading notifications"),
+		).toBeInTheDocument();
+
+		const retryButton = body.getByRole("button", { name: /retry/i });
+		await userEvent.click(retryButton);
+		await expect(
+			body.queryByText("Error loading notifications"),
+		).not.toBeInTheDocument();
+	},
+};
+
+export const MarkAllAsRead: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: fn(() =>
+			Promise.resolve({ notifications: MockNotifications, unread_count: 2 }),
+		),
+		markAllAsRead: fn(() => Promise.resolve()),
+	},
+	play: async ({ canvasElement }) => {
+		const body = within(canvasElement.ownerDocument.body);
+		let unreads = await body.findAllByText(/unread/i);
+		await expect(unreads).toHaveLength(2);
+		const markAllAsReadButton = body.getByRole("button", {
+			name: /mark all as read/i,
+		});
+
+		await userEvent.click(markAllAsReadButton);
+		unreads = body.queryAllByText(/unread/i);
+		await expect(unreads).toHaveLength(0);
+	},
+};
+
+export const MarkAllAsReadFailure: Story = {
+	decorators: [withGlobalSnackbar],
+	args: {
+		defaultOpen: true,
+		fetchNotifications: fn(() =>
+			Promise.resolve({ notifications: MockNotifications, unread_count: 2 }),
+		),
+		markAllAsRead: fn(() =>
+			Promise.reject(
+				mockApiError({ message: "Failed to mark all notifications as read" }),
+			),
+		),
+	},
+	play: async ({ canvasElement }) => {
+		const body = within(canvasElement.ownerDocument.body);
+		const markAllAsReadButton = body.getByRole("button", {
+			name: /mark all as read/i,
+		});
+		await userEvent.click(markAllAsReadButton);
+		await body.findByText("Failed to mark all notifications as read");
+	},
+};
+
+export const MarkNotificationAsRead: Story = {
+	args: {
+		defaultOpen: true,
+		fetchNotifications: fn(() =>
+			Promise.resolve({ notifications: MockNotifications, unread_count: 2 }),
+		),
+		markNotificationAsRead: fn(() =>
+			// true as true is necessary to solve a really strange TypeScript error
+			// https://stackoverflow.com/questions/75864591/type-boolean-is-not-assignable-to-type-true
+			Promise.resolve({ is_read: true as true }),
```

Comment:
and then all of this goes away

```suggestion
			Promise.resolve(),
```